src/material/button/_button-theme.scss

```diff
@@ -27,21 +27,36 @@ $_mat-button-ripple-opacity: 0.1;
   }
 }
 
+// If the $background-color resolved to something different from a color (e.g. a CSS variable),
+// We need to apply opacity explicitly. Because we cannot add opacity to CSS variable with "rgba" function
+// called inside "mat-color" function
+@mixin _mat-button-ripple-apply-background-opacity-if-needed($background-color, $opacity) {
+  @if (type-of($background-color) != color) {
+    opacity: $opacity;
+  }
+}
+
 @mixin _mat-button-ripple-color($theme, $hue, $opacity: $_mat-button-ripple-opacity) {
   $primary: map-get($theme, primary);
   $accent: map-get($theme, accent);
   $warn: map-get($theme, warn);
 
   &.mat-primary .mat-ripple-element {
-    background-color: mat-color($primary, $hue, $opacity);
+    $background-color: mat-color($primary, $hue, $opacity);
+    background-color: $background-color;
+    @include _mat-button-ripple-apply-background-opacity-if-needed($background-color, $opacity)
```

Comment:
This could be just a bit more concise. I would do this
```scss
// Applies the background color for a ripple. If the value provided is not a Sass color,
// we assume that we've been given a CSS variable. Since we can't perform alpha-blending
// on a CSS variable, we instead add the opacity directly to the ripple element.
@mixin _mat-button-ripple-background($palette, $hue, $opacity) {
  $background-color: mat-color($palette, $hue, opacity);
  background-color: $background-color;
  @if (type-of($background-color) != color) {
    opacity: $opacity;
  }
}

&.mat-primary .mat-ripple-element {
  @include _mat-button-ripple-background($primary, $hue, $opacity);
}

```

Reply:
Thanks for your advice, I made another update.